pkg/tsdb/v3iotsdb_integration_test.go

```diff
@@ -0,0 +1,350 @@
+	}
+}
+
+func setUp(t *testing.T, v3ioConfig *config.V3ioConfig) func() {
+	dbConfig := config.DBPartConfig{
+		DaysPerObj:     1,
+		HrInChunk:      1,
+		DefaultRollups: "sum",
+		RollupMin:      10,
+	}
+
+	if err := CreateTSDB(v3ioConfig, &dbConfig); err != nil {
+		t.Fatalf("Failed to create TSDB. reason: %s", err)
+	}
+
+	return func() {
+		deleteTSDB(t, v3ioConfig)
+	}
+}
+
+func setUpWithDBConfig(t *testing.T, v3ioConfig *config.V3ioConfig, dbConfig config.DBPartConfig) func() {
+	if err := CreateTSDB(v3ioConfig, &dbConfig); err != nil {
+		t.Fatalf("Failed to create TSDB. reason: %s", err)
+	}
+
+	return func() {
+		deleteTSDB(t, v3ioConfig)
+	}
+}
+
+func TestIngestData(t *testing.T) {
+	v3ioConfig, err := config.LoadConfig(filepath.Join("../../", config.DefaultConfigurationFileName))
+	if err != nil {
+		t.Fatalf("Failed to load test configuration. reason: %s", err)
+	}
+
+	fmt.Printf("som %v", testUtils.DataPoint{Time: 123134, Value: 123.43})
+
+	testCases := []struct {
+		desc       string
+		metricName string
+		labels     []utils.Label
+		data       []testUtils.DataPoint
+	}{
+		{desc: "Should ingest one data point", metricName: "cpu", labels: utils.FromStrings("testLabel", "balbala"),
+			data: []testUtils.DataPoint{{Time: 1532940510, Value: 314.3}}},
+
+		{desc: "Should ingest multiple data points", metricName: "cpu",
+			labels: utils.FromStrings("os", "linux", "iguaz", "yesplease"),
+			data: []testUtils.DataPoint{{Time: 1532940510, Value: 314.3},
+				{Time: 1532940510 + 5, Value: 300.3},
+				{Time: 1532940510 + 10, Value: 3234.6}}},
+
+		{desc: "Should ingest record with late arrival", metricName: "cpu",
+			labels: utils.FromStrings("os", "linux", "iguaz", "yesplease"),
+			data: []testUtils.DataPoint{{Time: 1532940510, Value: 314.3},
+				{Time: 1532940510 + 5, Value: 300.3},
+				{Time: 1532940510 - 10, Value: 3234.6}}},
+
+		{desc: "Should ingest record with '-' in the metric name (IG-8585)", metricName: "cool-cpu",
+			labels: utils.FromStrings("os", "linux", "iguaz", "yesplease"),
+			data: []testUtils.DataPoint{{Time: 1532940510, Value: 314.3},
+				{Time: 1532940510 + 5, Value: 300.3},
+				{Time: 1532940510 - 10, Value: 3234.6}}},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.desc, func(t *testing.T) {
+			testIngestDataCase(t, v3ioConfig, test.metricName, test.labels, test.data)
+		})
+	}
+}
+
+func testIngestDataCase(t *testing.T, v3ioConfig *config.V3ioConfig,
+	metricsName string, userLabels []utils.Label, data []testUtils.DataPoint) {
+	defer setUp(t, v3ioConfig)()
+
+	adapter, err := NewV3ioAdapter(v3ioConfig, nil, nil)
+	if err != nil {
+		t.Fatalf("Failed to create v3io adapter. reason: %s", err)
+	}
+
+	appender, err := adapter.Appender()
+	if err != nil {
+		t.Fatalf("Failed to get appender. reason: %s", err)
+	}
+
+	labels := utils.Labels{utils.Label{Name: "__name__", Value: metricsName}}
+	labels = append(labels, userLabels...)
+
+	fmt.Printf("the labeles are: %v\n", labels)
```

Comment:
Typo. Do we really want to print stuff here? And if so, shouldn't we use the logger?